coupling_components/solver_wrappers/abaqus/USRInit.f

```diff
@@ -90,24 +98,31 @@ SUBROUTINE DLOAD(F,KSTEP,KINC,TIME,NOEL,NPT,LAYER,KSPT,
       ID = 0
 #endif
 
-      R = INDEX(SNAME,'SURFACE')
-      READ(SNAME((R+7):LEN(TRIM(SNAME))),'(I)') R
-      R = R+1
+      IF (S > 1) THEN
+         DO R = 1,S
+            IF (INDEX(SNAME, TRIM(SURFACEIDS(R))) > 0) THEN
+               EXIT
+            END IF
+         END DO
+      ELSE
+         R = 1
+      END IF
```

Comment:
Should we include an error message if a match is not found? From what I see here in case no match is found, R will take the value of S so the USRInit.f (and the entire initialize-method in python) will run. However, the model parts from the fluid an structure side will not match and give an error later on in the code without being clear what the cause is.

Reply:
I agree and pushed some changes that implement this. I also included an error message when the names are substrings of each other.